src/php.php

```diff
@@ -0,0 +1,84 @@
+<?php
+function getConnected() : ?PDO
+{
+    try
+    {
+        $connection = new PDO(
+            'mysql:host=DB;dbname=collectorapp',
+            'root',
+            'password'
+        );
+        $connection->setAttribute(PDO:: ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+        return $connection;
+    } catch (PDOException $exception)
+    {
+        echo 'connection error: ' . $exception->getMessage();
+        return null ;
+    }
+}
+
+function fetchPigmentData() : array
+{
+    // Establish a database connection
+    $connection = getConnected();
+
+    try {
+        $query = $connection->prepare('SELECT pigment.id, pigment.name, pigment.HEX, pigment.mineral, pigment.chemical, pigment.description, pigment.image_closeup, color.name AS color_name
+            FROM pigment
+            INNER JOIN color color ON pigment.color_id = color.color_id');
+        $query->execute();
+        return $query->fetchALL();
+
+    } catch (PDOException $exception){
+        // In case of a database-related error, catch the exception and print an error message
+        echo 'fetch error:' . $exception->getMessage();
+        // Return an empty array if there's an error
+        return[];
+    }
+}
+function createpigmaDIVS(array $collection = null): string
```

Comment:
I'm not a big fan of this function name for a few reasons:
* 'pigma' seems like a shortened word
* 'pigma' is not snake_cased OR camelCased to distinguish it as a word
* DIVS is in ALLCAPS but it isn't an acronym
* 'create pgima divs' is close to being a technical detail - `displayPigments` would be a more conventional name that tells you *what the function does* and not *how it does it* - in programming, the *how* things are done often changes as an application is developed, but the 'what' rarely changes.